cluster-api-gcp/cloud/google/machineactuator.go

```diff
@@ -490,8 +490,7 @@ func (gce *GCEClient) requiresUpdate(a *clusterv1.Machine, b *clusterv1.Machine)
 		!reflect.DeepEqual(a.Spec.Roles, b.Spec.Roles) ||
 		!reflect.DeepEqual(a.Spec.Versions, b.Spec.Versions) ||
 		a.ObjectMeta.Name != b.ObjectMeta.Name ||
-		a.ObjectMeta.UID != b.ObjectMeta.UID ||
-		!reflect.DeepEqual(aConfig, bConfig)
```

Comment:
I might be missing something here but how are the unit tests passing with the provider config comparison removed?